src/harness/testcases/WINNF_FT_S_WDB_testcase.py

```diff
@@ -205,24 +246,48 @@ def test_WINNF_FT_S_WDB_2(self, config_filename):
     # Send registration requests.
     registration_responses = self._sas.Registration(registration_requests)['registrationResponse']
 
-    # Check registeration response,
+    # Check registration response,
     # responseCode should be 103 (INVALID_VALUE).
     for registration_response in registration_responses:
       self.assertEqual(registration_response['response']['responseCode'], 103)
 
     del registration_responses
 
-    # TODO
     # Step 2: Create a CPI database which includes CPI user with credentials
     # matching those used to create the request in Step 1.
+    cpi_database = DatabaseServer('CPI Database',config['cpiDatabaseConfig']['hostName'],
+                                  config['cpiDatabaseConfig']['port'])
+
+    # Start CPI database server.
+    cpi_database.start()
+
+    # Generate the CSV file and write the content of it.
+    config_keys = config['cpiDatabaseConfig']['cpis'][0].keys()
+
+    with open(config['cpiDatabaseConfig']['cpis'][0]['publicKeyIdentifierFile'], 'w+') as output_file:
```

Comment:
This is the wrong file. You should be writing index.csv (which is then used on line 275), not overwriting the PKI files.

Please store the path in a local var and use that var below, to reduce the chance for error.

Please call the file index.csv.